test/test_nestedtensor.py

```diff
@@ -6050,6 +6050,49 @@ def test_copy_(self, device):
         ):
             a.copy_(b)
 
+    def test_index_put_(self, device):
+        offsets = torch.tensor([0, 2, 5, 7], device=device)
+        a = torch.nested.nested_tensor_from_jagged(
+            torch.zeros(7, 3, device=device), offsets
+        )
+        indices = [
+            torch.tensor([0, 1, 2], device=a.device),
+            torch.tensor([0, 2, 1], device=a.device),
+            torch.tensor([0, 0, 0], device=a.device),
+        ]
+        a[indices] = 1.0
+
+        sentences = a.unbind()
+        self.assertEqual(sentences[0][0], torch.tensor([1.0, 0.0, 0.0]))
+        self.assertEqual(sentences[0][1], torch.tensor([0.0, 0.0, 0.0]))
+        self.assertEqual(sentences[1][2], torch.tensor([1.0, 0.0, 0.0]))
+
+        import subprocess
+
+        with self.subTest():
+            r = subprocess.call(
+                [
+                    sys.executable,
+                    "-c",
+                    """\
+import torch
+offsets = torch.tensor([0, 2, 5, 7], device='cuda')
+lengths = torch.tensor([2, 2, 2], device='cuda')
+indices = [
+    torch.tensor([0, 1, 2], device='cuda'),
+    torch.tensor([0, 2, 1], device='cuda'),
+    torch.tensor([0, 0, 0], device='cuda'),
+]
+a = torch.nested.nested_tensor_from_jagged(
+    torch.zeros(7, 3, device='cuda'), offsets, lengths
+)
+a[indices] = 1.0
+torch.cuda.synchronize()
+""",
+                ]
+            )
+            self.assertTrue(r != 0)
```

Comment:
I'm a bit confused - is this call expected to fail? the args seem okay to me.

Reply:
check me on this, but I expect all zeros with the following locations as ones. There should be 3 updates, since `len(indices[0]) == len(indices[1]) == len(indices[2]) == 3`:
* `(batch=0, ragged=0, embedding=0)` AKA `(0, 0, 0)`
* `(batch=1, ragged=2, embedding=0)` AKA `(1, 2, 0)`
* `(batch=2, ragged=1, embedding=0)` AKA `(2, 1, 0)`

the nt has shape `(B, S*, D) == (3, [2, 3, 2], 3)` so I'd expect all the above to be updated successfully

Reply:
this is a non-contiguous NJT with holes, the nt has shape `(B, S*, D) == (3, [2, *2*, 2], 3)`!

Reply:
yep that's what I was missing, thanks :)